starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -66,4 +67,47 @@ public void create() {
     Assert.assertTrue(inventoryPostInsertion.size() == existingInventory.size() + 1);
     Assert.assertTrue(createdInventory.getId().equals(returnedInventory.getId()));
   }
+  
+  /**
+   * Test Delete method with valid id.
+   */
+  @Test
+  public void delete() {
+    // Put an inventory in the databse so there's something to delete
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    // Save the id upon insertion to check against when we delete it
+    Inventory savedInventory = mongoTemplate.save(inventory);
+    String targetId = savedInventory.getId();
+    // Count number of inventories in collection to make sure it's 1 less after a deletion
+    List<Inventory> existingInventories = this.mongoTemplate.findAll(Inventory.class);
+    // Delete an inventory, and check id of deleted inventory against the id we saved earlier
+    Optional<Inventory> deletedInventory = this.inventoryDAO.delete(targetId);
+    // Check size of collection post deletion
+    List<Inventory> inventoryPostDeletion = this.mongoTemplate.findAll(Inventory.class);
+    // Check that id of deleted inventory matches our target id
+    Assert.assertTrue(deletedInventory.map(Inventory::getId).orElse("default").equals(targetId));
+    // Check that size of collection has decreased by 1
+    Assert.assertTrue(inventoryPostDeletion.size() == existingInventories.size() - 1);
+  }
+
+  /**
+   * Test Delete method with invalid id
+   */
+  @Test
+  public void deleteAndProvideInvalidId() {
+    // Put an inventory in the collection so there's something to delete
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    Inventory savedInventory = mongoTemplate.save(inventory);
+    // Count number of inventories in collection to make sure it's unchanged after this deletion attempt
+    List<Inventory> existingInventories = this.mongoTemplate.findAll(Inventory.class);
+    // Attempt deletion but provide invalid id
+    Optional<Inventory> deletedInventory = this.inventoryDAO.delete("wombat");
```

Comment:
How do you know that the id "wombat" isn't the id for inventory? The id for inventory isn't defined.. it could be anything

Reply:
Hey Ben, so I now realize that I was confused. I wrote this comment as a sort of follow-on to the comment that the id should still be set even if it's going to change (which I now take back), and not so much about the invalid id string itself. You don't need to set the id in these test cases since you are creating new test inventories for each test. If you were creating test inventories for all tests to use, then you would need to define a test_id and set it, depending on the goal of the test. (I hope I didn't cause too much confusion, I'm just so used to seeing it done that way, that I didn't consider your way to also be valid, I'm sorry. So as you know mongo will generate an ObjectId when it gets inserted, so for testing the case of an invalid id - you should be able to use a random, non-empty string.

Reply:
Okydoky no sweat I changed it back (: